feature-libs/customer-ticketing/components/customer-ticketing-dialog/customer-ticketing-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import {
+  CustomerTicketingConfig,
+  MAX_ENTRIES_FOR_ATTACHMENT,
+  MAX_INPUT_CHARACTERS,
+  MAX_INPUT_CHARACTERS_FOR_SUBJECT,
+  MAX_SIZE_FOR_ATTACHMENT,
+} from '@spartacus/customer-ticketing/root';
+import {
+  FilesFormValidators,
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+
+@Directive()
+export abstract class CustomerTicketingDialogComponent {
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+  inputCharactersLimit: number = this.getInputCharactersLimit;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  get messagesCharacterLeft(): number {
+    return (
+      this.inputCharactersLimit - (this.form.get('message')?.value?.length || 0)
+    );
+  }
+
+  get allowedTypes(): string[] | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.allowedTypes;
+  }
+
+  get getInputCharactersLimit(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing?.inputCharactersLimit ||
+      MAX_INPUT_CHARACTERS
+    );
+  }
+
+  get getInputCharactersForSubject(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing
+        ?.inputCharactersLimitForSubject || MAX_INPUT_CHARACTERS_FOR_SUBJECT
+    );
+  }
+
+  get maxSize(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing?.attachmentRestrictions
+        ?.maxSize || MAX_SIZE_FOR_ATTACHMENT
+    );
+  }
+
+  get maxEntries(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing?.attachmentRestrictions
+        ?.maxEntries || MAX_ENTRIES_FOR_ATTACHMENT
+    );
+  }
+
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      this.close('Click outside of the window');
+    }
+  }
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected el: ElementRef,
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected filesFormValidators: FilesFormValidators
+  ) {}
+
+  close(reason: string): void {
+    this.launchDialogService.closeDialog(reason);
+  }
+}
```

Comment:
not related to this line

you can include an abstract method, so the extended class will have to implement the form


abstract buildForm(): void

Reply:
got it

Reply:
I added the whole method in abstract component instead as 2 fields are common in all components and that way only create component will need to add the form fields.